mobius-core/src/main/java/com/spotify/mobius/EffectRouterBuilder.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius;
+
+import com.spotify.mobius.functions.Consumer;
+
+/** TODO: document! */
+public interface EffectRouterBuilder<F, E> {
+
+  <G extends F> EffectRouterBuilder<F, E> add(Class<G> klazz, Runnable action);
```

Comment:
Should we include a `WorkRunner` to be used when running this particular action/consumer? This won't cover effects that need to be run on specific threads such as UI effects.

Reply:
You get a problem with the lifecycle of workrunners if you use them in a connectable router - basically you have to pass in a `Provider<WorkRunner>` that creates new runners every time the router is connected, otherwise you might end up with old (disposed) workrunners. I suggested to petter that maybe he should skip all WorkRunners in the connectable-router API and instead let rx och the runnable handles threads on their own instead.

Reply:
Lifecycle management is another reason for that - but I don't think that's fundamentally different from the Rx API? You could receive a `Scheduler` that is based on a thread pool that has been shut down or gets shut down later on, without the router being aware of that. If we did have `WorkRunner`s in this API, the lifecycle management of those runners would anyway have to be done separately from the connectable, I'd say.